### DISCUSSION TITLE
why doesn't create cookies in localhost after added firestore and SSR

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [gladly-team/next-firebase-auth#572] [Q&A]
**Strange bugs in localhost after SSG , ISR , Firestore in my device**

**Describe the bug**
First I connect firestore for data fetching after auth . then I write some SSR code and It works . then everything is crazy in Localhost .
1. it doesn't redirect after auth
2. it doesn't create cookies but it create firebase indexDB stuff
3. I can confirm it happens in localhost , also in [example app](https://github.com/gladly-team/next-firebase-auth/tree/v1.x/example) with same firebase config 
4. you can find my last (no error version) github file in [here](https://github.com/thanhtutzaw/too)
5. error happened when doing firestore , ssr and isr stuff in Localhost
6. now my whole device have this error . also in official next-firebase-auth [example app](https://github.com/gladly-team/next-firebase-auth/tree/v1.x/example) . I think I have problem in firebase account . But only error in localhost .

**Versions**

`next-firebase-auth` version: ^1.0.0-canary.9
Firebase JS SDK: 9.9
Next.js: 12.1

**Expected behavior**
It should be create cookies and redirect to app after auth .

**Debug and error logs**
Error: Received 500 response from login API endpoint: {"error":"Unexpected error when Login"}

FirebaseAuthError: `uid` argument must be a non-empty string uid.
    at FirebaseTokenGenerator.createCustomToken (D:\React\Next\too\node_modules\firebase-admin\lib\auth\token-generator.js:99:19)
    at Auth.createCustomToken (D:\React\Next\too\node_modules\firebase-admin\lib\auth\base-auth.js:88:36)
  errorInfo: {
    code: 'auth/argument-error',
    message: '`uid` argument must be a non-empty string uid.'
  },
  codePrefix: 'auth'
}

**Additional context**
in Network tab after logged in , login api is pending and after 20s it says status 500 error in login API endpoint.
![Screenshot (75)](https://user-images.githubusercontent.com/71011043/200124430-af4656ab-76af-4fd1-a0cb-8f82c6dfb65e.png)
![Screenshot (69)](https://user-images.githubusercontent.com/71011043/200124604-67c3cffd-bb9a-48cf-9bfe-6a13e2de77fb.png)

**Comment by kmjennison** [comment]
Please confirm you've done the following:

1. Have you set `onVerifyTokenError` and `onTokenRefreshError` in your config and checked for any helpful errors?
2. Have you set `debug: true` in your config and read through server-side and client-side debug logs for any helpful messages?
3. Have you tried the [example app](https://github.com/gladly-team/next-firebase-auth/tree/v1.x/example) with your own Firebase credentials?
4. Have you read through the [troubleshooting](https://github.com/gladly-team/next-firebase-auth/tree/v1.x#troubleshooting) Q&A?

**Reply by kmjennison** [reply]
If none of this helps, please provide a repository with a minimal reproduction of the issue.

**Reply by thanhtutzaw** [reply]
Can You check my github ([no error](https://github.com/thanhtutzaw/too)) repo with your firebase config ? If it works I have problem in My firebase . That hosted in netlify and env include in netlify . So it works in production. https://too.netlify.app . My problem exist in Local enviroment . I also disable cookie secure false in iniAuth config and this login api 500 is still happening and `FirebaseAuthError: `uid` argument must be a non-empty string uid.
`

**Reply by thanhtutzaw** [reply]
1. I don't get token error message but I got login api 500 error and uid empty error in ssr , client  auth is work . no redirect
2. yes debug is always true
3. I tried example app with too.firebase account , that doesn't work too. What is wrong with my account.
4. I saw Q & A and I tried cookie secure to false and Still don't work .

**Reply by kmjennison** [reply]
@thanhtutzaw Could you try the latest canary version of `next-firebase-auth`?

The example app works for me and other users, so it seems it's related to your configuration.  Please double-check your Firebase values and other settings. Unfortunately, I'm not able to help debug every implementation.

**Reply by thanhtutzaw** [reply]
I think it is problem in my PC .
I tried different browser, still doesn't work. I don't know why but something is happening. After I added Firestore, getting data with ssr , isr and this happened in my PC. I also tried with new firebase config and it doesn't work too. 
Very weird problem. The login api is pending and getting 500 status. Local Server is not working I think. Client is working .Also working well in production(netlify) in my machine.
